### ISSUE TITLE
Issue with Lmer function

### ISSUE THREAD

**Issue by rsuhendra** [ejolly/pymer4#105]
# Issue
There seems to be a problem with the Lmer function. I was having some installation issues (which I think I fixed) so I tried running the test_function:
python -c "from pymer4.test_install import test_install; test_install()"
and it gave me this error: 
Error! Length mismatch: Expected axis has 7 elements, new values have 2 elements

I tried it with all of the 4 installations:
1. `conda create --name test_env1 -c ejolly -c defaults -c conda-forge pymer4`
2. `conda create --name test_env2 -c ejolly -c conda-forge -c defaults pymer4`
3. `conda create --name test_env3 -c ejolly/label/pre-release -c defaults -c conda-forge pymer4`
4. `conda create --name test_env4 -c ejolly/label/pre-release -c conda-forge -c defaults pymer4`
and all had the same error. I've also tried this some other of my own conda environments on 3.7 and 3.8, but those didn't work either. 

I dug around in the code a little and it seems to be a problem with some tranposing of the data. Specifically, the error comes out on line 533: df.index = out_rownames (same on line 622, which came up when I was originally fitting to binomial data). 

I tried fixing it by adding the line 'df = df.T' before line 533, and it seems to work, however, another error pops up on  line 707: self.design_matrix.columns = self.coefs.index[:] . This error also goes away if I tranpose the design matrix, but then having "hotfixed" my way out of the other stuff, I got an error on line 721: ran_vars = df.query("var2 in @varcor_NAs").drop("var2", axis=1), which I didn't really know how to deal with (I also don't have this error on hand right now but I don't think it was the same issue). 

Anyways, here's a few specs:

[pymer4_issue.txt](https://github.com/ejolly/pymer4/files/8469904/pymer4_issue.txt)

% uname -mprsv
Darwin 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:47:26 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T8101 arm64 arm

% sw_vers
ProductName:	macOS
ProductVersion:	12.3.1
BuildVersion:	21E258

Let me know if the error is reproducible by anyone else, could definitely be something funky with how R on my conda installation or something.

**Comment by gpiantoni** [comment]
I bumped into the same issue as @rsuhendra running Linux, Python 3.9.12 packaged by conda-forge. I tried the same hotfixes that that user proposed but I hit the same block. 

The issue appeared when I upgraded rpy2 from 3.4.5-py39r41hce5d2b2_3 to 3.5.1-py39r41hd257fcd_0. I suspect the bug is somewhere in the conversion from r-objects to pandas DataFrame.

Downgrading to  3.4.5-py39r41hce5d2b2_3 is my temporary solution but I would appreciate a better solution. Thanks a lot!
